AutomatticTracks/src/main/java/com/automattic/android/tracks/crashlogging/internal/SentryErrorTrackerProxy.kt

```diff
@@ -0,0 +1,23 @@
+package com.automattic.android.tracks.crashlogging.internal
+
+import android.content.Context
+import io.sentry.SentryEvent
+import io.sentry.SentryOptions
+import io.sentry.protocol.User
+
+interface SentryErrorTrackerProxy {
```

Comment:
I'm wondering if we really need to introduce this interface. Can't we simply work with `SentryErrorTrackerProxyImpl`? 

1. The interface is dependant on Sentry anyway, so it won't help us if we ever decide to replace Sentry with another library.
2. Mockito allows us to mock even classes so this interface imo won't bring any value when it comes to testing.
3. Kotlin compiler is smart, so it won't re-compile files which don't need to be re-compiled (as oppose to Java compiler which afaik always recompiles all affected dependencies).

All in all I'm wondering if we are not just introducing an unnecessary boilerplate. Wdyt?

Reply:
It was introduced to make testing easier (creating a fake object). As I've followed your suggestion in https://github.com/Automattic/Automattic-Tracks-Android/pull/51#discussion_r603352435, I've refactored tests to use mocks so eventually I've removed this interface. 

Answering your questions:
1. Totally right, this interface didn't abstract idea of `ErrorTracker` in general
2. If we use `Mockito` - that's true. For `Fakes`, we wouldn't be able to write test
3. Are you referring to `compile avoidance`? This was not a reason for writing this interface, as Kotlin doesn't support it. Editing `SentryErrorTrackerProxy**Impl**` would cause recompilation of `CrashLogging` in Kotlin (wouldn't if it would be Java). Or do you refer to the incremental compilation? If so, I'm not sure if I understand how it works: maybe the internal implementation of `SentryErrorTrackerProxy**Impl**` changes won't cause recompilation of `CrashLogging`?

Reply:
> Are you referring to compile avoidance? This was not a reason for writing this interface, as Kotlin doesn't support it. Editing SentryErrorTrackerProxy**Impl** would cause recompilation of CrashLogging in Kotlin (wouldn't if it would be Java). Or do you refer to the incremental compilation? If so, I'm not sure if I understand how it works: maybe the internal implementation of SentryErrorTrackerProxy**Impl** changes won't cause recompilation of CrashLogging?

I don't have a deep understanding of compilers. I just remember a talk where they compared java vs kotlin compiler. One of the points was that introducing interfaces in java to speed up incremental compilation can help in Java but is not necessary in Kotlin, since kotlin compiler is smart enough.

> This was not a reason for writing this interface, as Kotlin doesn't support it.

Hmm, I didn't know that Kotlin doesn't support it. I've read [this article](https://blog.jetbrains.com/kotlin/2020/09/the-dark-secrets-of-fast-compilation-for-kotlin/) and it seems that it supports it. Or do you think it's different for java+kotlin projects? I'm just curious, feel free to ignore this comment if you don't know from the top of your head.